docs/en-us/dev/user_doc/alert_spi.md

```diff
@@ -0,0 +1,102 @@
+
+You can read the relevant code under the dolphinscheduler-spi module as a reference. The extended interface of the associated plug-in is under the module, and when we need to implement the relevant function plug-in, it is recommended to read this block first. Of course, you are advised to read the documentation, which will take a lot less time.  However, the documentation has a certain lag and when it is missing, it is suggested that the source code prevails (If you are interested, we also welcome you to submit relevant documents).  In addition, we make few changes (excluding additions) to the extension interface, except for major architectural changes where there is an incompatible upgrade version, so existing documentation is generally adequate.
+
+When you need to extend, you actually only need to focus on the extension interface. For example, for the alert service you only need to focus on the AlertChannelFactory and the AlertChannel. The underlying logic is already implemented by DolphinScheduler, which makes our development much more focused and simple.
+
+By the way, we use an excellent front-end component, form-create, which supports the generating of front-end ui components based on json. If plugin development involves a front-end, we will use json to generate the relevant front-end UI components. The parameters of the plugin are wrapped in org.apache.dolphinscheduler.spi.params, which converts all the relevant parameters into json, meaning that you can draw the front-end components (mainly forms here, we only care about the data for front and back-end interaction) in Java code.
+
+This article focuses on the design and development of Alert alerts.
+
+#### Main Modules
+
+If you don't care about its internal design, just want to know how to develop your own alarm plugin, you can skip this section.
+
+* dolphinscheduler-spi
+
+     This module is the core SPI module and provides the basic SPI relevant behaviour,where dolphinschedulerplugin is the plugin top layer interface, all DolphinsCheduler's plugins must implement the interface, and the module also provides some universal tool classes (if you can detach it again Some will something will be better? For example, the UI, we currently use the parameter this block) and some UI related basic information.
+
+* dolphinscheduler-alert
+
+     In this module, we have implemented the load of the associated plugin when the Alert-Server starts. Alert provides a variety of plug-in configuration methods that can be enabled by simple configurations after you have done a job. The configuration file is located at dolphinscheduler-alert/src/main/resources/alert.properties .
+
+     It provides two methods of configuration.

```

Comment:
Why is there a 'two methods' for 3 items?

Reply:
> Why is there a 'two methods' for 3 items?

This is also what I doubt, in the process of translation.